core/server/common/src/main/java/alluxio/master/journal/raft/transport/CopycatGrpcConnection.java

```diff
@@ -100,13 +107,19 @@
    * Note: {@link #setTargetObserver} should be called explicitly before using the connection.
    *
    * @param connectionOwner owner of connection
+   * @param transportId transport level Id
    * @param context copycat thread context
    * @param executor transport executor
    * @param requestTimeoutMs timeout in milliseconds for requests
    */
-  public CopycatGrpcConnection(ConnectionOwner connectionOwner, ThreadContext context,
-      ExecutorService executor, long requestTimeoutMs) {
+  public CopycatGrpcConnection(ConnectionOwner connectionOwner, String transportId,
+      ThreadContext context, ExecutorService executor, long requestTimeoutMs) {
     mConnectionOwner = connectionOwner;
+    mConnectionId = MoreObjects.toStringHelper(this)
+        .add("ConnectionOwner", mConnectionOwner)
+        .add("ConnectionId", sConnectionIdCounter.incrementAndGet())
```

Comment:
overflowing (resetting to 0) should not cause any issues?

Reply:
This is only for tracing. 
Also it would take decade for this one to overflow.